vault/activity_log.go

```diff
@@ -1698,12 +1699,40 @@ func (a *ActivityLog) handleQuery(ctx context.Context, startTime, endTime time.T
 			})
 		}
 	}
-
+	months = modifyResponseMonths(months, startTime, endTime)
 	responseData["months"] = months
 
 	return responseData, nil
 }
 
+// modifyResponseMonths fills out various parts of the query structure to help
+// activity log clients parse the returned query.
+func modifyResponseMonths(months []*ResponseMonth, start time.Time, end time.Time) []*ResponseMonth {
+	if len(months) == 0 {
+		return months
+	}
+	start = timeutil.StartOfMonth(start)
+	end = timeutil.EndOfMonth(end)
+	modifiedResponseMonths := make([]*ResponseMonth, 0)
+	firstMonth, err := time.Parse(time.RFC3339, months[0].Timestamp)
+	lastMonth, err2 := time.Parse(time.RFC3339, months[len(months)-1].Timestamp)
+	if err != nil || err2 != nil {
+		return months
+	}
+	for start.Before(firstMonth) {
+		monthPlaceholder := &ResponseMonth{Timestamp: start.UTC().Format(time.RFC3339)}
+		modifiedResponseMonths = append(modifiedResponseMonths, monthPlaceholder)
+		start = timeutil.StartOfMonth(start.AddDate(0, 1, 0))
+	}
+	modifiedResponseMonths = append(modifiedResponseMonths, months...)
+	for lastMonth.Before(end) {
```

Comment:
up until this point we've been building `modifiedResponseMonths` up in chronological order, but here we'll be going in reverse for the months past what's included in `months`

e.g. if `months` goes from june-aug of <some_year>, with `start` being january and `end` being december, this would look like:

[jan feb mar apr may june july aug dec nov oct sept]

is that a problem? e.g. should we instead count from the end of `months` up until `end`, resulting in:

[jan feb mar apr may june july aug sept oct nov dec]

Reply:
whoops! didn't mean to un-resolve 🥴 was just checking the status